### PR TITLE
Support OS-Agent Data disk

### PR DESCRIPTION
**Pull request by pvizeli**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/1056
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/api/os.py [hunk 1]
```diff
@@ -29,22 +32,30 @@ class APIOS(CoreSysAttributes):
...
-            ATTR_VERSION: self.sys_hassos.version,
-            ATTR_VERSION_LATEST: self.sys_hassos.latest_version,
-            ATTR_UPDATE_AVAILABLE: self.sys_hassos.need_update,
-            ATTR_BOARD: self.sys_hassos.board,
+            ATTR_VERSION: self.sys_os.version,
+            ATTR_VERSION_LATEST: self.sys_os.latest_version,
+            ATTR_UPDATE_AVAILABLE: self.sys_os.need_update,
+            ATTR_BOARD: self.sys_os.board,
             ATTR_BOOT: self.sys_dbus.rauc.boot_slot,
+            ATTR_DISK_DATA: self.sys_os.datadisk.disk_used,
```
Comment (ludeeus) [inline supervisor/api/os.py:40]: Changes to the API need a documentation change.

File: supervisor/jobs/const.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ class JobCondition(str, Enum):
     INTERNET_HOST = "internet_host"
     RUNNING = "running"
     HAOS = "haos"
+    HOST_AGENT = "host_agent"
```
Comment (ludeeus) [inline supervisor/jobs/const.py:21]: Let's use  OS_AGENT = "os_agent"

Comment (ludeeus) [reply supervisor/jobs/const.py:21]: Should do that for `HostFeature` as well

File: supervisor/os/disk.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+"""Home Assistant Operating-System DataDisk."""
```
Comment (ludeeus) [inline supervisor/os/disk.py:1]: We might need other disk related files in the future.
Let's use `data_disk.py` for the filename to match the class here.

File: supervisor/os/disk.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        # Need some error handling, but we need know what disk_used will return
+        try:
+            await self.sys_dbus.agent.datadisk.change_device(new_disk)
+        except DBusError as err:
+            raise HassOSError(
+                f"Can't move data partition to {new_disk!s}: {err!s}", _LOGGER.error
+            ) from err
+
+        try:
+            await self.sys_host.control.reboot()
```
Comment (ludeeus) [inline supervisor/os/disk.py:42]: Can the system be used at all after `self.sys_dbus.agent.datadisk.change_device` is called?
Or are a reboot always needed directly after?

File: supervisor/utils/json.py [hunk 5]
```diff
@@ -25,9 +25,7 @@ def default(self, o: Any) -> Any:
         if isinstance(o, set):
             return list(o)
         if isinstance(o, Path):
-            return str(o)
-        if hasattr(o, "as_dict"):
```
Comment (ludeeus) [inline supervisor/utils/json.py:29]: Why remove `as_dict` ?

Comment (pvizeli) [reply supervisor/utils/json.py:29]: because we are not the Core?

File: supervisor/os/disk.py (hunk 4 above)
Comment (pvizeli) [reply supervisor/os/disk.py:42]: > are a reboot always needed directly after?
yes

**Review by ludeeus — APPROVED** [review]

**Pull request merged**